__tests__/index.js

```diff
@@ -0,0 +1,1057 @@
+// 			& {
+// 				color:red;
+// 			}
+// 		`)).toBe(`.user{color:blue;}.user{color:red;}`)
+// })
+
+// TODO: original has kept some comments, not sure which and if it's even desired
+test("comments", () => {
+  expect(
+    defaultStylis(`
+			// line comment
+			// color: red;
+			/**
+			 * removes block comments and line comments,
+			 * there's a fire in the house // there is
+			 */
+			button /*
+			 	// what's
+			 	xxx
+			  */
+			{color: blue;}
+			// hello
+			button /* 1 */
+			{
+				color: red; /* 2 */
+			}
+			/*! 1 */
+			color: red;
+			/*! 2 */
+			h1 {
+				/*! 1 */
+				color: red;
+				/*! 2 */
+				color: red;
+				/*! 3 */
+			}
+		`)
+  ).toBe(
+    [
+      `.user{color:red;}`,
+      `.user button{color:blue;}.user button{color:red;}`,
+      `.user h1{color:red;color:red;}`
+    ].join("")
+  );
+});
+
+// TODO: doesn't work because compile doesn't replace & correctly
+// test('&', () => {
+// 	expect(defaultStylis(`
+// 			& {
+// 				color:blue;
+// 			}
+// 			&&& {
+// 				color:red;
+// 			}
+// 			& + & {
+// 				color:red;
+// 			}
+// 			.wrapper button& {
+// 			  color:red;
+// 			}
+// 			:hover & {
+// 			  color: green;
+// 			}
+// 			div:hover & {
+// 			  color: green;
+// 			}
+// 			div:hover & {
+// 				h1 & {
+// 					color:red;
+// 				}
+// 			}
+// 		`)).toBe([
+// 			`.user{color:blue;}`,
+// 			`.user.user.user{color:red;}`,
+// 			`.user + .user{color:red;}`,
+// 			`.wrapper button.user{color:red;}`,
+// 			`.user:hover .user{color:green;}`,
+// 			`div:hover .user{color:green;}`,
+// 			`h1 div:hover .user{color:red;}`
+// 		].join(''))
+// })
+
+// TODO: doesn't work because compile doesn't replace & correctly
+// test('&:before', () => {
+// 	expect(defaultStylis(`
+// 			&:before{
+// 				color:blue;
+// 			}
+// 		`)).toBe(`.user:before{color:blue;}`)
+// })
+
+// TODO: @import doesn't get compiled correctly (it doesn't appear in compiled output at all, even without wrapping input in namespace)
+// test('@import', () => {
+// 	expect(defaultStylis(`@import url('http://example.com')`)).toBe(`@import url('http://example.com')`)
+// })
+
+// TODO: declarations in at-rule are not grouped in a rule
+// TODO: original output was also auto-prefixed
+// test('@supports', () => {
+// 	expect(defaultStylis(`
+// 			@supports (display:block) {
+// 				color:red;
+// 				h1 {
+// 					color:red;
+// 					h2 {
+// 						color:blue;
+// 					}
+// 				}
+// 				display:none;
+// 			}
+// 			@supports (appearance: none) {
+// 				color:red;
+// 			}
+// 			@supports (backdrop-filter: blur(10px)) {
+// 				backdrop-filter: blur(10px);
+// 			}
+// 		`)).toBe([
+// 			'@supports (display:block){.user{color:red;display:none;}.user h1{color:red;}.user h1 h2{color:blue;}}',
+// 			'@supports (appearance:none){.user{color:red;}}',
+// 			'@supports (backdrop-filter:blur(10px)){.user{backdrop-filter:blur(10px);}}'
+// 		].join(''))
+// })
+
+// TODO: declarations in at-rule are not grouped in a rule
```

Comment:
I've done some digging and compared it to how the same thing gets done without at-rule context. A rule gets added here:
https://github.com/thysultan/stylis.js/blob/c0b8dc3ae17c7ee88710caae29993a10c394aa5f/src/Ruleset.js#L70
but the namespace for `@media` is not  added here, so when declarations in a ruleset are processed they land directly as children of a `@media`.

Because the rule is created here proactively we could probably just create a rule inside of that `@media` proactively as well and it would fix the issue. But the fact the rule gets created proactively makes stylis output empty rules at the moment (it's a possibility when parsing smth like this `namespace{.cls{color:hotpink}}` ) - question is if it's good to create empty rules and just leave it for a serialization to omit them or should we aim for not creating empty rules in the first place?